raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -0,0 +1,216 @@
+
+    def has_ready_channel(self, partner_address: Address) -> bool:
+        if partner_address not in self.address_to_rtc_partners:
+            return False
+        channel = self.address_to_rtc_partners[partner_address].channel
+        if channel is None:
+            return False
+        if channel.readyState == "open":
+            return True
+        return False
+
+    def async_create_channel(self, partner_address: Address) -> None:
+        assert self.node_address, "Transport is not started yet but tried to create rtc channel"
+        rtc_partner = self.get_rtc_partner(partner_address)
+        spawn_coroutine(
+            coroutine=create_channel(
+                rtc_partner, self.node_address, self._handle_message_callback
+            ),
+            callback=self._handle_sdp_callback,
+            partner_address=partner_address,
+        )
+
+    def async_set_remote_description(
+        self, partner_address: Address, description: Dict[str, str]
+    ) -> None:
+        assert self.node_address, "Transport is not started yet but tried to set candidates"
+        rtc_partner = self.get_rtc_partner(partner_address)
+        spawn_coroutine(
+            coroutine=set_remote_description(
+                rtc_partner=rtc_partner,
+                node_address=self.node_address,
+                description=description,
+                handle_message_callback=self._handle_message_callback,
+            ),
+            callback=self._handle_sdp_callback,
+            partner_address=partner_address,
+        )
+
+    def stop(self) -> None:
+        if self.node_address is None:
+            return
+
+        log.debug("Gracefully closing rtc channels", node=to_checksum_address(self.node_address))
+
+        for rtc_partner in self.address_to_rtc_partners.values():
+            if rtc_partner.channel:
+                rtc_partner.channel.close()
+
+
+async def create_channel(
+    rtc_partner: RTCPartner, node_address: Address, handle_message_callback: Callable
+) -> RTCSessionDescription:
+
+    pc = rtc_partner.peer_connection
+    rtc_partner.create_channel(node_address)
+
+    offer = await pc.createOffer()
+    await pc.setLocalDescription(offer)
+
+    log.debug("Created offer", offer=offer)
+
+    msg = "Channel must be created already"
+    assert rtc_partner.channel is not None, msg
+
+    @rtc_partner.channel.on("message")
+    async def on_message(message: Dict[str, bytes]) -> None:  # pylint: disable=unused-variable
+        log.debug(
+            "Received message in asyncio kingdom",
+            node=to_checksum_address(node_address),
+            message=message,
+            time=time.time(),
+        )
+        handle_message_callback(message, rtc_partner.partner_address)
+
+    @rtc_partner.channel.on("close")
+    def on_close() -> None:  # pylint: disable=unused-variable
+        rtc_partner.channel = None
+
+    return pc.localDescription
+
+
+async def set_remote_description(
```

Comment:
`async def set_remote_description(...)`
vs.
`def async_set_remote_description(...)` (in `WebRTCManager`)

That's not confusing at all 😉